src/Npgsql/Internal/Postgres/DataTypeName.cs

```diff
@@ -226,7 +226,7 @@ internal static string NormalizeName(string dataTypeName)
     }
 
     public override string ToString() => Value;
-    public bool Equals(DataTypeName other) => !IsDefault && !other.IsDefault && _value == other._value;
+    public bool Equals(DataTypeName other) => string.Equals(_value, other._value);
```

Comment:
Wonder if we can get away with ordinal comparer here...

Reply:
This is the ordinal comparer, and it's faster than passing it in explicitly as the method doesn't have to do validation of the enum (check both overloads)

Reply:
Out of interest: why didn't you use `_value == other._value`? It's the same comparison.
Though the `string.Equals` as used is more explicit. Was that the reason?

Reply:
Yes, mostly to be more explicit.